pkg/scheduler/schedule_one.go

```diff
@@ -735,11 +743,8 @@ func findNodesThatPassExtenders(ctx context.Context, extenders []framework.Exten
 				// note that this only happens if the extender returns the node in both maps
 				continue
 			}
-			if _, found := statuses[failedNodeName]; !found {
-				statuses[failedNodeName] = framework.NewStatus(framework.Unschedulable, failedMsg)
-			} else {
-				statuses[failedNodeName].AppendReason(failedMsg)
-			}
+			aggregatedReasons := aggregateReasons(failedNodeName, failedMsg)
+			statuses.Set(failedNodeName, framework.NewStatus(framework.Unschedulable, aggregatedReasons...))
```

Comment:
This is an existing issue unrelated to this PR though, what if extender-1 sets UnschedulableAndUnresolvable and then extender-2 sets Unschedulable afterwards? 
Should we not overwrite the status with `Unschedulable`, but only overwrite the reason with aggregated one?

Reply:
I think this scenario is not possible. `extender.Filter` returns list of feasible nodes ([ref](https://github.com/kubernetes/kubernetes/blob/8e0f2e83498d91b698b548ca165a0a9880f18fc5/pkg/scheduler/schedule_one.go#L727)), which denotes on which nodes the pod can be scheduled (after this filter). The `feasibleNodes` list is updated after each filter ([ref](https://github.com/kubernetes/kubernetes/blob/8e0f2e83498d91b698b548ca165a0a9880f18fc5/pkg/scheduler/schedule_one.go#L751)), so the next `extender.Filter` will get nodes that are neither `Unschedulable` nor `UnschedulableAndUnresolvable`.

If extender-1 will set `UnschedulableAndUnresolvable` for some node, this node won't be passed to the extender-2, so it won't get overwritten in the `NodeToStatus` as well.

Reply:
> If extender-1 will set UnschedulableAndUnresolvable for some node, this node won't be passed to the extender-2, so it won't get overwritten in the NodeToStatus as well.

..wait, then why do we need to agrregate reasons? Could one Node get more than one reason?

- `findNodesThatPassExtenders()` receives Nodes that didn't get rejected at PreFilter/Filter phase.
- Once an extender rejects a Node, the Node isn't passed to a next extender.

Reply:
That's right! None of `feasibleNodes` can has a status in `statuses` in each step in `findNodesThatPassExtenders`. It is sufficient just to write the status into map without any redundant aggregation.